### ISSUE TITLE
[Bug Report] Player is never buffering

### ISSUE THREAD

**Issue by clragon** [alexmercerind/media_kit#227]
I am unable to get into a state where the player reports that it is buffering, even when the video is being loaded.

I have tested this issue in the `media_kit_test` application.
The following setup:

replace native sources with uncached links with large videos:

`sources_native.dart`
```dart
final sources = <String>[
  'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4',
  'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4',
];

Future<void> prepareSources() async {
  return;
}
```

add a stream to the seekbar:

`widgets.dart`
```dart
// inside _SeekBarState
bool hasStreamBuffered = false;
bool hasStateBuffered = false;

// inside _SeekBarState.initState subscriptions.AddAll
widget.player.streams.buffering.listen((event) {
  setState(() {
    if (event) {
      hasStreamBuffered = true;
    }
    if (widget.player.state.buffering) {
      hasStateBuffered = true;
    }
  });
}),
```

add a widget displaying whether `buffering` was _ever_ true:

`widgets.dart`
```dart
// inside _SeekBarState.build
Row(
  children: [
    hasStreamBuffered ? const Icon(Icons.stream) : const SizedBox(),
    hasStateBuffered
        ? const Icon(Icons.data_array)
        : const SizedBox(),
  ],
),
```

I have additionally tried to print out these booleans, but they are never `true`.

From what I can tell, buffering _should_ have been true at least for a short while, since when I click into an "unbuffered" part of the video, there is a short pause, where the player is loading that part. This does not seem to happen.

**Comment by alexmercerind** [comment]
Hi! The `buffering` behavior has now been fixed once & for all.

The important update is that [`package:media_kit` is now 1.0.0](https://github.com/alexmercerind/media_kit/releases/tag/media_kit-v1.0.0). A lot of small quirks & issues have been fixed. Now all platforms are supported; including web. The [unit-tests have been vastly improved](https://github.com/alexmercerind/media_kit/tree/main/media_kit/test/src) in order to:
- Improve the stability
- Parity between platforms
- Avoiding regressions.

The major PR finally landed (#250).

<hr>

Please refer to the updated README for details; a new guide section has also been added.

**Comment by alexmercerind** [comment]
Please don't hesitate to report any issues :)
